bokehjs/src/lib/models/tools/gestures/range_tool.ts

```diff
@@ -72,8 +76,8 @@ export namespace RangeTool {
   export type Attrs = p.AttrsOf<Props>
 
   export type Props = Tool.Props & {
-    x_range: p.Property<Range1d | null>
-    y_range: p.Property<Range1d | null>
+    x_range: p.Property<Range | null>
+    y_range: p.Property<Range | null>
```

Comment:
No issues with `FactorRange` ?

Reply:
No issues (I tested), because all panning and zooming is based on `Range.{start,end}` properties. With this change it's easy to share one of the ranges by using e.g. `RangeTool(x_range=plot.x_range)`.